src/services/StatisticsCollectorImpl.groovy

```diff
@@ -50,54 +64,56 @@ class StatisticsCollectorImpl implements StatisticsCollector {
     private int getNumberOfMergeConflicts(Project project, MergeCommit mergeCommit) {
         int numberOfMergeConflicts = 0
 
-        Process gitShow = ProcessRunner.runProcess(project.getPath(), 'git', 'show', mergeCommit.getSHA())
-        try {
-            BufferedReader reader = new BufferedReader(new InputStreamReader(gitShow.getInputStream()))
-            ArrayList<String> output = reader.readLines()
+        simulateMergeScenario(project, mergeCommit)
 
-            for(line in output) {
-                if(line.endsWith("======="))
-                    numberOfMergeConflicts++
-            }
+        String mergeFiles = ProcessRunner.runProcess(project.getPath(), 'git', 'diff').getText()
+        List<String> list = new ArrayList<String>()
+        Pattern pattern = Pattern.compile(CONFLICT_MARKER)
+        Matcher m = pattern.matcher(mergeFiles)
+       
+        while (m.find()) {
+            numberOfMergeConflicts++
+        }
 
-        } catch(IOException e) {
-            e.printStackTrace()
-        }        
+        returnToMaster(project)
 
         return numberOfMergeConflicts
     }
 
     private int getNumberOfConflictingFiles(Project project, MergeCommit mergeCommit) {
-        int numberOfConflictingFiles = 0
-
-        Process gitShow = ProcessRunner.runProcess(project.getPath(), 'git', 'show', mergeCommit.getSHA())
-        try {
-            BufferedReader reader = new BufferedReader(new InputStreamReader(gitShow.getInputStream()))
-            ArrayList<String> output = reader.readLines()
-
-            boolean fileHasConflict = false
-            for(line in output) {
-                
-                if(line.startsWith('diff --cc') && fileHasConflict) {
-                    numberOfConflictingFiles++
-                } else if(line.startsWith('diff --cc')) {
-                    fileHasConflict = false
-                }
+        Set<String> conflictingFiles = new HashSet<String>()
 
-                if(line.endsWith("======="))
-                    fileHasConflict = true
-            }
+        String mergeOutput = simulateMergeScenario(project, mergeCommit).getText()
 
-            if(fileHasConflict)
-                numberOfConflictingFiles++
+        def lines = mergeOutput.split("\n")
+        Pattern pattern = Pattern.compile(CONFLICT_INDICATOR)
 
-        } catch(IOException e) {
-            e.printStackTrace()
-        }        
+        for (line in lines) {
+            Matcher m = pattern.matcher(line)
+            if (m.find())
+                conflictingFiles.add(line)
+        }
+
+        returnToMaster(project)
+        return conflictingFiles.size()
+    }
+    
+    private Process simulateMergeScenario(Project project, MergeCommit mergeCommit) {
+        Process checkoutLeft = ProcessRunner.runProcess(project.getPath(), 'git', 'checkout', mergeCommit.getLeftSHA())
```

Comment:
Um potencial problema aqui: eu acho, mas não tenho certeza, que se o master do repositório contiver mudanças não commitadas (se for local), o git não vai deixar o checkout acontecer.
Você poderia confirmar isso e, se for caso, pode levantar uma exceção ou usar stash/stash-apply pra evitar isso.

Reply:
@pauloborba pode tirar essa dúvida?

Reply:
Realmente, a solução depende do comportamento desejado pra ele, eu acho melhor levantar a exceção pra deixar essa escolha pro usuário.